ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -175,16 +175,17 @@ export class AzureTreeDataProvider<TRoot = ISubscriptionRoot> implements IAzureT
         }
     }
 
-    public async showTreeItemPicker(expectedContextValues: string | string[], startingTreeItem?: AzureTreeItem<TRoot | ISubscriptionRoot>): Promise<AzureTreeItem<TRoot | ISubscriptionRoot>> {
+    public async showTreeItemPicker(expectedContextValues: string | (string | RegExp)[] | RegExp, startingTreeItem?: AzureTreeItem<TRoot | ISubscriptionRoot>): Promise<AzureTreeItem<TRoot | ISubscriptionRoot>> {
         if (!Array.isArray(expectedContextValues)) {
             expectedContextValues = [expectedContextValues];
         }
 
         // tslint:disable-next-line:strict-boolean-expressions
         let treeItem: AzureTreeItem<TRoot | ISubscriptionRoot> = startingTreeItem || await this.promptForRootTreeItem(expectedContextValues);
-        while (!expectedContextValues.some((val: string) => treeItem.contextValue === val)) {
-            if (treeItem instanceof AzureParentTreeItem) {
-                treeItem = await treeItem.pickChildTreeItem(expectedContextValues);
+
+        while (!expectedContextValues.some((val: string | RegExp) => (val instanceof RegExp && treeItem.contextValue.match(val) !== null) || treeItem.contextValue === val)) {
+            if ((<AzureParentTreeItem>treeItem).hasMoreChildrenImpl) {
```

Comment:
I'm fine with this change, but add a comment saying why we're not using `instanceof`

Reply:
Also - I think I'd prefer to check `if ((<AzureParentTreeItem>treeItem).loadMoreChildrenImpl) {` just because `hasMoreChildrenImpl` already returns a boolean and that seems potentially more confusing